simulator/src/framebuffer.rs

```diff
@@ -0,0 +1,143 @@
+use crate::{display::SimulatorDisplay, output_settings::OutputSettings};
+use embedded_graphics::{
+    drawable::{Drawable, Pixel},
+    geometry::{Point, Size},
+    pixelcolor::{PixelColor, Rgb888, RgbColor},
+    primitives::{Primitive, Rectangle},
+    style::{PrimitiveStyle, Styled},
+    DrawTarget,
+};
+use std::convert::TryFrom;
+
+/// Rgb888 framebuffer
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Framebuffer {
+    size: Size,
+    pub(crate) data: Box<[u8]>,
```

Comment:
I assume this is here for good reason, but why can't it be a `&'a [u8]`?

Reply:
`Framebuffer` needs to own data. If a slice would be used the `new` method wouldn't be able to allocate `data`, because it would only be valid inside the `new` method.

Reply:
Ah yeah sure. This can be a `Vec<u8>` if it's owned then, no?

Reply:
Would that have any advantage? I did choose a boxed slice because `data` is a fixed size array and shouldn't change in size during the lifetime of a `Framebuffer`.

Reply:
Nah I guess not, the `Box<>` just looks weird :)